tests/execute/basic/test.sh

```diff
@@ -4,14 +4,10 @@
 rlJournalStart
     function check_duration() {
         local result_file=$1
-        local results=$2
-        DURATION_REGEXP="^(((([0-1][0-9])|(2[0-3])):?[0-5][0-9]:?[0-5][0-9]+$))"
-        DURATION=$(grep -A5 "$results" "$result_file" | grep "duration:" | awk '{print $2}')
-        if [[ "$DURATION" =~ $DURATION_REGEXP ]]; then
-          rlRun "true" 0 "duration is in HH:MM:SS format"
-        else
-          rlRun "false" 0 "duration isn't in HH:MM:SS format"
-        fi
+        local test_name=$2
+
+        rlRun "yq -r '.[] | select(.name == \"$test_name\") | .duration | test(\"^(((([0-1][0-9])|(2[0-3])):?[0-5][0-9]:?[0-5][0-9]+$))\")'" \
```

Comment:
We're using `yq` here so this means the pull request depends on 1849?

Reply:
definitely, your insight to the comments would be apprecated there

Reply:
If we are on it. Is the regex correct? The seconds specification matches also `00:11:2222` and similar? There seems to be a redundant `+` sign?

Reply:
Also, are the optional colons (the `:?` parts) correct? It allows situations like these, which look weird to me:
* `10:1010` - matched
* `10:110` - not matched

Reply:
`10:10` is not according to specs, so should not be accepted

Reply:
I did not touch the regular expression, I merely moved it around. It should be as it was - if it's incorrect, I'd rather leave fixing it to another patch.

Reply:
How about `^(?:(?:[01][0-9])|(?:2[0-3])):[0-5][0-9]:[0-5][0-9]$`? I'll fix the pattern here because I need it for schema anyway.

Reply:
Modified in 7b3d570c

Reply:
The pattern looks better. Not sure about the hours though: Probably not that common, but I guess `duration` exceeding one day like `44:33:22` is still valid, right?

Reply:
Hmm, you're right, that's absolutely allowed. So `[0-9][0-9]` for hours?

Reply:
I'd say even `[0-9]+` would be fine to allow `103:11:22`.

Reply:
`[0-9]{2,}` seems to be the winner.

Reply:
Could it be as simple as `^[0-9]{2,}:[0-5][0-9]:[0-5][0-9]$` ?

Reply:
Treid in 7362bc09

Reply:
Yes, sounds good to me. Unless we are ok with `1:20:40`, that is using a single digit hour, as well.

Reply:
I'd not allow `1:...`, the specification says `hh`. `[0-9]{2,}` should require at least two digits, so we should be fine.

Reply:
Ok, good.